### DISCUSSION TITLE
Why is the bundle size that large?

### DISCUSSION THREAD

**Discussion by EfstathiadisD** [tailwindlabs/headlessui#568] [General]
I am looking into the Lib, and I am currently kind of amazed of the bundle size of this lib. Why is it so large? Can we have tree-shakable components? Instead of installing the whole thing, have `import {Popover} from '@headlessui/react/popover'`?

**Comment by maplessmann** [comment]
Looks like tree-shaking is supported after this PR https://github.com/tailwindlabs/headlessui/pull/602
